components/generic_modal.scss

```diff
@@ -98,6 +98,152 @@
         border: none;
         border-radius: 4px;
     }
+
+    &.GenericModal__compassDesign {
+        .modal-content {
+            padding: 0;
+            border-radius: 12px;
+
+            .modal-header {
+                padding: 26px 32px 15px;
+
+                .close {
+                    top: 24px;
+                    right: 22px;
+                    width: 30px;
+                    height: 30px;
+                    box-shadow: none;
+
+                    &:active {
+                        background: rgba(var(--button-bg-rgb), 0.08);
+                        color: var(--button-bg);
+                    }
+
+                    &:focus {
+                        box-sizing: border-box;
+                        border: 2px solid var(--sidebar-text-active-border);
+                    }
+                }
+
+                .GenericModal__header {
+                    padding: 0;
+                    border-top-left-radius: 12px;
+                    border-top-right-radius: 12px;
+
+                    h1 {
+                        color: var(--center-channel-color);
+                        font-size: 22px;
+                    }
+                }
+            }
+
+            .modal-body {
+                padding: 15px 32px 0;
+
+                .GenericModal__body {
+                    padding: 0;
+                }
+            }
+
+            .modal-footer {
+                padding: 0 32px 26px;
+                border-bottom-left-radius: 12px;
+                border-bottom-right-radius: 12px;
+            }
+
+            .genericModalError {
+                display: flex;
+                box-sizing: border-box;
+                flex: 1;
+                padding: 14px;
+                border: 1px solid rgba(247, 67, 67, 0.16);
```

Comment:
I'm making one comment here but this should apply to all colors: we should try to use variables from `/sass/base/_css_variables.scss` whenever possible for colors, for example this is one is `--dnd-indicator-rgb`. Maybe check with @matthewbirtch if we should not use the `--error-text-color-rgb` for this one tho - as it would make more sense to have all errors using the same color.